src/libstore/s3_binary_cache_store.go

```diff
@@ -0,0 +1,131 @@
+package libstore
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/minio/minio-go/v7"
+	"io"
+	"log"
+	"net/url"
+	"os"
+)
+
+type S3Store struct {
+	Profile    string
+	Region     string
+	Scheme     string
+	bucketName string
+	client     *minio.Client
+}
+
+// SDK should be in environment
+func (s *S3Store) getUri() string {
+	return "s3://" + s.bucketName
+}
+
+func (s *S3Store) FileExists(ctx context.Context, path string) (bool, error) {
+	found, err := s.client.BucketExists(ctx.Background(), s.bucketName)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	if found {
+		obj, err := s.client.GetObject(ctx, s.bucketName, path, minio.GetObjectOptions{})
+		if err != nil {
+			log.Fatalln(err)
+			return
+		}
+		defer reader.Close()
+
+		localFile, err := os.Create("my-testfile")
+		if err != nil {
+			log.Fatalln(err)
+		}
+		defer localFile.Close()
+
+		stat, err := reader.Stat()
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		if _, err := io.CopyN(localFile, reader, stat.Size); err != nil {
+			log.Fatalln(err)
+			return
+		}
+	}
+
+	return false
```

Comment:
did you build that code? this part shouldn't compile as it's missing the error part

Reply:
> did you build that code? this part shouldn't compile as it's missing the error part

I modified some test in the library.